engine/src/mac-window.mm

```diff
@@ -453,29 +453,42 @@ - (void)windowDidDeminiaturize:(NSNotification *)notification
 }
 
 // MW-2014-07-22: [[ Bug 12720 ]] Mark the period we are inside a focus event handler.
+// SN-2015-05-20: [[ Bug 15208 ]] Renamed to better reflect the functions action
 - (void)windowDidBecomeKey:(NSNotification *)notification
 {
     if (s_inside_focus_event)
-        m_window -> ProcessDidBecomeKey();
+        m_window -> ProcessGainedMainFocus();
     else
     {
         s_inside_focus_event = true;
-        m_window -> ProcessDidBecomeKey();
+        m_window -> ProcessGainedMainFocus();
         s_inside_focus_event = false;
     }
 }
 
 // MW-2014-07-22: [[ Bug 12720 ]] Mark the period we are inside a focus event handler.
+// SN-2015-05-20: [[ Bug 15208 ]] If we are not the KeyWindow, we can still be
+//  the MainWindow, and in that case we keep the focus.
 - (void)windowDidResignKey:(NSNotification *)notification
 {
-    if (s_inside_focus_event)
-        m_window -> ProcessDidBecomeKey();
-    else
-    {
-        s_inside_focus_event = true;
-        m_window -> ProcessDidResignKey();
-        s_inside_focus_event = false;
-    }
+}
+
+// SN-2015-05-20: [[ Bug 15208 ]] A main window is not necessarily the key window
```

Comment:
The same happens when using any of the answer or ask dialogs, have they a text field or not.